### PR TITLE
feat(releases): migrate releases from gitlab to github

### PR DESCRIPTION
**Pull request by gaurav5430**
Added code to transfer releases from gitlab to github with limited functionality (due to github api limitations).

### CODE REVIEW

File: src/githubHelper.ts [hunk 1]
```diff
@@ -160,6 +160,62 @@ export default class GithubHelper {
...
 
+  
+  /**
+   * Gets a release by tag name
+   * @param tag {string} - the tag name to search a release for
+   * @returns 
+   */
+   async getReleaseByTag(tag) {
+    try {
+      await utils.sleep(this.delayInMs);
```
Comment (spruce) [inline src/githubHelper.ts:171]: Is the sleep here neccessary?

File: src/githubHelper.ts [hunk 2]
```diff
@@ -160,6 +160,62 @@ export default class GithubHelper {
...
+   * @param tag_name {string} - the tag name
+   * @param name {string} - title of the release
+   * @param body {string} - description for the release
+   */
+   async createRelease(
+    tag_name: string, 
+    name: string,
+    body: string) {
+    try {
+      await utils.sleep(this.delayInMs);
```
Comment (spruce) [inline src/githubHelper.ts:199]: Is the sleep here neccessary?

File: src/index.ts [hunk 3]
```diff
@@ -149,6 +156,8 @@ async function migrate() {
         await transferMergeRequests();
       }
     }
+
+
```
Comment (spruce) [inline src/index.ts:160]: Could you remove these newlines? :)

File: src/index.ts [hunk 4]
```diff
@@ -447,6 +456,63 @@ async function transferMergeRequests() {
...
+        await githubHelper.createRelease(release.tag_name, release.name, release.description);
+      } catch (err) {
+        console.error(
+          'Could not create release: !' +
+          release.name + ' - ' + release.tag_name
+        );
+        console.error(err);
+      }
+    } else {
+      if (githubRelease) {
```
Comment (spruce) [inline src/index.ts:506]: This if is always true because it is in the else branch of the inverse and can therefor be removed.

**Review by spruce — CHANGES_REQUESTED** [review]
Looks good sofar. But I found some things which I would like to be changed.
Also, please add a comment about the new feature to sample settings and the README.

Thanks 👍

File: src/githubHelper.ts (hunk 1 above)
Comment (gaurav5430) [reply src/githubHelper.ts:171]: it seems like this is a pattern we are following whenever we make the GitHub api call, to make sure that we do not exceed the quota. I have added this based on my understanding of the existing code

File: src/index.ts (hunk 3 above)
Comment (gaurav5430) [reply src/index.ts:160]: would do

File: src/index.ts (hunk 4 above)
Comment (gaurav5430) [reply src/index.ts:506]: would remove

File: src/githubHelper.ts (hunk 2 above)
Comment (gaurav5430) [reply src/githubHelper.ts:199]: same as above

https://github.com/piceaTech/node-gitlab-2-github/pull/94#discussion_r763250387

**Comment by gaurav5430** [comment]
> Looks good sofar. But I found some things which I would like to be changed. Also, please add a comment about the new feature to sample settings and the README.
> 
> Thanks 👍

- added to README and sample settings
- replied to PR comments
- fixed whitespace

File: src/index.ts (hunk 3 above)
Comment (gaurav5430) [reply src/index.ts]: removed

File: src/index.ts (hunk 4 above)
Comment (gaurav5430) [reply src/index.ts]: removed

File: src/githubHelper.ts (hunk 1 above)
Comment (spruce) [reply src/githubHelper.ts:197]: For now we can leave it in. I just merged PRs relating to quatas / abuse limits. If it turns out it makes the migration slower than needed we will remove it later.

**Pull request merged**